src/Dns/Dns/Zones/NewAzureDnsZone.cs

```diff
@@ -101,8 +114,107 @@ public override void ExecuteCmdlet()
                     this.WriteVerbose(zoneType == Management.Dns.Models.ZoneType.Private
                         ? string.Format(ProjectResources.Success_NewPrivateZone, this.Name, this.ResourceGroupName)
                         : string.Format(ProjectResources.Success_NewZone, this.Name, this.ResourceGroupName));
+
                     this.WriteObject(result);
+
+                    DnsZone parent = this.ParseParentZoneFromArguments();
+                    if (parent != null && this.Name.EndsWith(parent.Name))
+                    {
+                        AddDnsNameserverDelegation(result, parent);
+                        this.WriteVerbose(ProjectResources.Success);
+                        this.WriteVerbose(string.Format(ProjectResources.Success_NSDelegation, this.Name, parent.Name));
+                    }
                 });
         }
+        /// <summary>
+        /// This method adds name server delegation in the parent zone for the created child zone.
+        /// The NS records are added in the parent zone that corresponds to the child zone
+        /// </summary>
+        /// <param name="zone">the created child zone</param>
+        /// <param name="parent">the parent zone</param>
+        private DnsRecordSet AddDnsNameserverDelegation(DnsZone zone, DnsZone parent)
+        {
+            DnsRecordSet recordSet = null;
+            if (zone != null && parent != null && zone.NameServers != null && zone.NameServers.Count > 0)
+            {
+                List<NsRecord> nameServersList = new List<NsRecord>();
+                foreach (string nameserver in zone.NameServers)
+                {
+                    NsRecord record = new NsRecord();
+                    record.Nsdname = nameserver;
+                    nameServersList.Add(record);
+                }
+
+                DnsRecordBase[] resourceRecords = nameServersList.ToArray();
+                string recordName = this.Name.Replace('.' + parent.Name, "");
+                recordSet = this.DnsClient.CreateDnsRecordSet(
+                    parent.Name,
+                    this.ResourceGroupName,
+                    recordName,
+                    3600,
+                    RecordType.NS,
+                    null,
+                    true,
+                    resourceRecords,
+                    null);
+            }
+            return recordSet;
+        }
+
+        /// <summary>
+        /// This method parses the parent zone from the command arguments, depending
+        /// on the format of the parent zone details passed.
+        /// Supports resource id, parent zone name and the zone object.
+        /// </summary>
+        private DnsZone ParseParentZoneFromArguments()
```

Comment:
@ShwethaRaghupathi if you'd like, we have our own resource id parsing class [here](https://github.com/Azure/azure-powershell-common/blob/master/src/ResourceManager/Version2016_09_01/Utilities/Models/ResourceIdentifier.cs) that you can use to grab certain parts of the id rather than having to re-implement the below logic. Your project already has a reference to the package containing this class, so you'd just need to use the namespace.

Reply:
Updated PR with this change.